src/audio/dai.c

```diff
@@ -841,6 +841,14 @@ static int dai_copy(struct comp_dev *dev)
 	if (!copy_bytes && dd->start_position != dev->position)
 		return 0;
 
+	/* Check possibility of glitch occurrence */
+	if (dev->direction == SOF_IPC_STREAM_PLAYBACK && copy_bytes + free_bytes < dd->period_bytes)
```

Comment:
I must admit I don't understand the math here. Adding free_bytes seems very odd to me?

Reply:
My intention was to compare number of bytes in DMA (after copy) with bytes eaten in on period, to assure DMA won't be hungry at the end of following period.